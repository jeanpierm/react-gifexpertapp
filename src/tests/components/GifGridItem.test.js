import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Pruebas en <GifGridItem/>", () => {
  const title = "Goku";
  const url = "https://localhost/goku.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  test("should show <GifGridItem/> correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe tener un párrafo con el título", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("debe tener la imagen los props url y title como src y alt respectivamente", () => {
    const img = wrapper.find("img");
    const { src, alt } = img.props();
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("debe tener animate__fadeIn", () => {
    const div = wrapper.find("div");
    const className = "animate__fadeIn";
    expect(div.hasClass(className)).toBe(true);
  });
});
