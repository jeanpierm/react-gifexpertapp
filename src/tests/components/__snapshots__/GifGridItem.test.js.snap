// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> should show <GifGridItem/> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Goku"
    src="https://localhost/goku.jpg"
  />
  <p>
    Goku
  </p>
</div>
`;
